src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -177,12 +181,22 @@ public async Task SendRequestBodyAsync(CancellationToken cancellationToken)
                     {
                         Debug.Assert(_requestCompletionState == StreamCompletionState.InProgress, $"Request already completed with state={_requestCompletionState}");
 
-                        _requestCompletionState = StreamCompletionState.Failed;
-
                         // Cancel above should ensure that the response is either Completed or Failed now.
                         Debug.Assert(_responseCompletionState != StreamCompletionState.InProgress);
 
-                        Reset();
+                        if (_requestBodyAbandoned)
+                        {
+                            // See comments on _requestBodyAbandoned.
+                            // In this case, the request is still considered successful and we do not want to send a RST_STREAM.
+                            Debug.Assert(_responseCompletionState == StreamCompletionState.Completed);
+                            _requestCompletionState = StreamCompletionState.Completed;
+                            Complete();
```

Comment:
Should we be eating the exception in this case?  It looks like it's still going to propagate out to the caller; what will happen then?

Reply:
Good question. I think we should eat the exception here. In particular, for non-duplex scenarios, we don't want to propagate the error.